### DISCUSSION TITLE
DuckDB shouldn't drop error messages from S3 when writing

### DISCUSSION THREAD

**Discussion by rmoff** [duckdb/duckdb#6594] [Ideas]
When using DuckDB to write data to S3-compatible endpoint it drops error detail and instead returns a generic error message. This makes it difficult to diagnose the actual issue that caused the failure. An example of this can be see in [this issue](https://github.com/duckdb/duckdb/issues/5685), as well as this which I hit: 

```
D export database 's3://drones03/main/drone-registrations/' (format parquet);
 94% ▕████████████████████████████████████████████████████████▍   ▏ Error: Invalid Error: IO Error: Unexpected response during S3 multipart upload finalization
```

(In this case the S3 endpoint was the lakeFS gateway, which returns a rich error message that would tell the user the cause of the error; `cannot write to protected branch`)

Related code: https://github.com/duckdb/duckdb/blob/1bb227b4107e18ccc02dbfe0ead0c701cbc92b6a/extension/httpfs/s3fs.cpp#L439-L442

**Comment by Mause** [comment]
I have a PR open for this presently: https://github.com/duckdb/duckdb/pull/6533

**Reply by rmoff** [reply]
Great to hear @Mause! :)
